app/controllers/katello/api/v2/host_subscriptions_controller.rb

```diff
@@ -35,6 +37,42 @@ def events
       respond_for_index :collection => collection
     end
 
+    api :POST, "/hosts/subscriptions/register", N_("Simulate a host subscription-manager registration")
+    param :name, String, :desc => N_("Name of the content host"), :required => true
+    param :uuid, String, :desc => N_("UUID to use for registered host")
+    param :facts, Hash, :desc => N_("Key-value hash of content host-specific facts") do
+      param :fact, String, :desc => N_("Any number of facts about this content host")
+    end
+    param :type, String, :desc => N_("Type of the content host, it should always be 'system'")
+    param :guest_ids, Array, :desc => N_("IDs of the virtual guests running on this content host")
+    param :installed_products, Array, :desc => N_("List of products installed on the content host")
+    param :release_version, String, :desc => N_("Release version of the content host")
+    param :service_level, String, :allow_nil => true, :desc => N_("A service level for auto-healing process, e.g. SELF-SUPPORT")
+    param :organization_id, :number, :desc => N_("Specify the organization"), :required => true
+    param :lifecycle_environment_id, String, :desc => N_("Specify the lifecycle environment"), :required => true
+    param :content_view_id, String, :desc => N_("Specify the content view"), :required => true
+    def register
```

Comment:
If this our supported API, I feel like we should follow standard CRUD rules then and not have a special "register" action to create a host with subscription aspect.